### PR TITLE
Add missing output types in TS API #9699

### PR DESCRIPTION
**Pull request by anatol-sialitski**

### CODE REVIEW

**Review by edloidas — APPROVED** [review]

File: modules/lib/lib-auth/src/main/resources/lib/xp/auth.ts [hunk 1]
```diff
@@ -774,7 +774,7 @@ interface CreateRoleHandler {
 
     setDescription(value?: string | null): void;
 
-    createRole(): Role;
+    createRole(): Role | null;
```
Comment (rymsha) [inline modules/lib/lib-auth/src/main/resources/lib/xp/auth.ts:777]: fix java instead

File: modules/lib/lib-content/src/main/resources/lib/xp/content.ts [hunk 2]
```diff
@@ -1102,7 +1102,7 @@ interface ModifyContentHandler {
  *
  * @returns {object} Modified content as JSON.
  */
-export function modify(params: ModifyContentParams): Content {
+export function modify(params: ModifyContentParams): Content | null {
```
Comment (rymsha) [inline modules/lib/lib-content/src/main/resources/lib/xp/content.ts:1105]: fix java instead

File: modules/lib/lib-content/src/main/resources/lib/xp/content.ts [hunk 3]
```diff
@@ -1181,7 +1181,7 @@ interface PublishContentHandler {
  *
  * @returns {object} Status of the publish operation in JSON.
  */
-export function publish(params: PublishContentParams): PublishContentResult {
+export function publish(params: PublishContentParams): PublishContentResult | null {
```
Comment (rymsha) [inline modules/lib/lib-content/src/main/resources/lib/xp/content.ts:1184]: fix java instead

File: modules/lib/lib-i18n/src/main/resources/lib/xp/i18n.ts [hunk 4]
```diff
@@ -45,7 +45,7 @@ interface LocaleScriptBean {
  *
  * @returns {string} The localized string.
  */
-export function localize(params: LocalizeParams): string {
+export function localize(params: LocalizeParams): string | null {
```
Comment (rymsha) [inline modules/lib/lib-i18n/src/main/resources/lib/xp/i18n.ts:48]: NOT_TRANSLATED_MESSAGE

File: modules/lib/lib-repo/src/main/resources/lib/xp/repo.ts [hunk 5]
```diff
@@ -143,7 +143,7 @@ interface CreateRepositoryHandler {
  * @returns {object} Repository created as JSON.
  *
  */
-export function create(params: CreateRepositoryParams): object {
+export function create(params: CreateRepositoryParams): Repository | null {
```
Comment (rymsha) [inline modules/lib/lib-repo/src/main/resources/lib/xp/repo.ts:146]: fix java code

File: modules/lib/lib-repo/src/main/resources/lib/xp/repo.ts [hunk 6]
```diff
@@ -253,7 +253,7 @@ interface CreateBranchHandler {
  * @return {object} The branch (as JSON).
  *
  */
-export function createBranch(params: CreateBranchParams): BranchResult {
+export function createBranch(params: CreateBranchParams): BranchResult | null {
```
Comment (rymsha) [inline modules/lib/lib-repo/src/main/resources/lib/xp/repo.ts:256]: fix java code

File: modules/lib/lib-repo/src/main/resources/lib/xp/repo.ts [hunk 7]
```diff
@@ -289,7 +289,7 @@ interface DeleteBranchHandler {
  * @return {object} The branch (as JSON).
  *
  */
-export function deleteBranch(params: DeleteBranchParams): BranchResult {
+export function deleteBranch(params: DeleteBranchParams): BranchResult | null {
```
Comment (rymsha) [inline modules/lib/lib-repo/src/main/resources/lib/xp/repo.ts:292]: fix java code

**Review by rymsha — CHANGES_REQUESTED** [review]

File: modules/lib/lib-content/src/main/resources/lib/xp/content.ts (hunk 2 above)
Comment (anatol-sialitski) [reply modules/lib/lib-content/src/main/resources/lib/xp/content.ts:1105]: Actually no, null is a possible value

**Review by rymsha — APPROVED** [review]

**Pull request merged**